web/home/src/pages/Home/home.tsx

```diff
@@ -67,82 +71,103 @@ function Home() {
-        let combinedPlugins: PluginHomePage[] = [
-          ...fakePluginsList,
-          ...res,
-        ];
+  // Keep track of the previous fetch
+  const [previousFetch, setPreviousFetch] = useState<PluginHomePage[]>([]);
+  const displayPlugins = async () => {
+    await getPlugins()
+      .then((res: PluginHomePage[]) => {
+        let combinedPlugins: PluginHomePage[] = [...fakePluginsList, ...res];
         // If the list of plugins has changed, update the state
         if (JSON.stringify(previousFetch) !== JSON.stringify(res)) {
           setPlugins(combinedPlugins);
-          previousFetch = res;
+          setPreviousFetch(res);
         }
       })
       .catch((err) => {
         console.log(err);
         // If there is an error, use the previous list
-        setPlugins(previousFetch);
+        setPlugins((previousFetch) =>
+          previousFetch.length > 0 ? previousFetch : fakePluginsList,
+        );
       });
-
+  };
+  // fetch plugins every 10 seconds
+  useEffect(() => {
+    // fetch plugins
+    const fetchData = async () => {
+      await displayPlugins();
+    };
+    fetchData();
+    const interval = setInterval(() => {
+      displayPlugins();
     }, 10000);
 
     return () => clearInterval(interval);
   }, []);
 
-const mapPluginList = () => {
-  return plugins.map((plugin) => {
-    if (plugin.status == PluginStatus.Starting || plugin.status == PluginStatus.Up){
-      return (
-        <PluginCard
-          {...{
-            plugin: {
-              id: plugin.id,
-              logo: plugin.logo ? plugin.logo : massaLogomark,
-              name: plugin ? plugin.name : 'Plugin Problem',
-              description: plugin.description
-                ? plugin.description
-                : 'Plugin Problem',
-              status: plugin.status ? plugin.status : 'Plugin Problem',
-            },
-            handleOpenPlugin: handleOpenPlugin,
-            key: plugin.id,
-          }}
-        />
-      );
-    }
-    else{
-      return (<></>)
-    }
-  })
-
-}
-
-return (
-  <div
-    className=" min-h-screen bg-img"
-    style={{ backgroundImage: `url(${grid1})` }}
-  >
-    <Header />
-
-    <MainTitle title="Which Plugins" />
+  const mapPluginList = () => {
+    return plugins
+      .filter((p) => !!p.name)
+      .sort((a, b) => a.name.localeCompare(b.name))
+      .map((plugin) => {
+        if (
+          plugin.status == PluginStatus.Starting ||
+          plugin.status == PluginStatus.Up
+        ) {
+          return (
+            <PluginCard
+              {...{
+                plugin: {
+                  id: plugin.id,
+                  logo: plugin.logo ? plugin.logo : massaLogomark,
+                  name: plugin ? plugin.name : 'Plugin Problem',
+                  description: plugin.description
+                    ? plugin.description
+                    : 'Plugin Problem',
+                  status: plugin.status ? plugin.status : 'Plugin Problem',
```

Comment:
Why Plugin Problem ?

Reply:
If it's displaying this , the plugin doesn't share the infos and it's a problem , i don't know what to display in this case to show to the user there is a problem